scripts/issue/474/upgrade_arb_crv_univ3.py

```diff
@@ -0,0 +1,117 @@
+from brownie import chain, interface
+from rich.console import Console
+
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import ADDRESSES_ARBITRUM
+from helpers.addresses import registry
+
+console = Console()
+
+DEV_MULTI = registry.arbitrum.badger_wallets.dev_multisig
+PROXY_ADMIN = registry.arbitrum.proxyAdminDev
+
+TRICRYPTO_STRATEGY = ADDRESSES_ARBITRUM["strategies"]["native.tricrypto"]
+TRICRYPTO_LOGIC = ADDRESSES_ARBITRUM["logic"]["native.tricrypto"]
+TRICRYPTO_VAULT = ADDRESSES_ARBITRUM["sett_vaults"]["bcrvTricrypto"]
+
+RENCRV_STRATEGY = ADDRESSES_ARBITRUM["strategies"]["native.renCrv"]
+RENCRV_LOGIC = ADDRESSES_ARBITRUM["logic"]["native.renCrv"]
+RENCRV_VAULT = ADDRESSES_ARBITRUM["sett_vaults"]["bcrvRenBTC"]
+
+UNIV3_ROUTER = ADDRESSES_ARBITRUM["uniswap"]["routerV3"]
+
+
+def main(simulation="true"):
+    safe = GreatApeSafe(DEV_MULTI)
+
+    console.print(f"Updating Tricrypto strategy logic to {TRICRYPTO_LOGIC}")
+    upgrade_strategy_logic(
+        TRICRYPTO_STRATEGY,
+        TRICRYPTO_LOGIC,
+        safe,
+        simulation,
+    )
+
+    console.print(f"Updating Rencrv strategy logic to {RENCRV_LOGIC}")
+    upgrade_strategy_logic(RENCRV_STRATEGY, RENCRV_LOGIC, safe, simulation)
+
+    safe.post_safe_tx(call_trace=True)
+
+
+def upgrade_strategy_logic(
+    strategy_proxy_address: str,
+    logic_address: str,
+    safe: GreatApeSafe,
+    simulation: bool = True,
+):
+    proxy_admin = interface.IProxyAdmin(PROXY_ADMIN, owner=safe.account)
+    strat_proxy = interface.ICrvStrategy(strategy_proxy_address, owner=safe.account)
+
+    prev_strategist = strat_proxy.strategist()
+    prev_controller = strat_proxy.controller()
+    prev_gov = strat_proxy.governance()
+    prev_guardian = strat_proxy.guardian()
+    prev_keeper = strat_proxy.keeper()
+    prev_perFeeG = strat_proxy.performanceFeeGovernance()
+    prev_perFeeS = strat_proxy.performanceFeeStrategist()
+    prev_reward = strat_proxy.reward()
+    prev_unit = strat_proxy.uniswap()
+    prev_gauge = strat_proxy.gauge()
+    prev_gauge_factory = strat_proxy.gaugeFactory()
+    prev_swapr_router = strat_proxy.SWAPR_ROUTER()
+
+    if simulation == "true":
+        chain.snapshot()
+
+        # Harvest on old strat, store gain in want
+        gauge = strat_proxy.gauge()
+        want = interface.ERC20(gauge)
+        prev_want_bal = want.balanceOf(strat_proxy.address)
+
+        chain.sleep(60 * 60 * 2)
```

Comment:
```suggestion
        chain.sleep(60 * 60 * 20)
```

Reply:
so weird, i wonder why? it has to be something underlying with brownie chain state and caching?